### PR TITLE
Update to Anthropic's claude-v2

### PR DESCRIPTION
**Pull request by mikelambert**

### CODE REVIEW

**Review by hinthornw — APPROVED** [review]

**Comment by rlancemartin** [comment]
May force users to upgrade Anthropic (not necessarily a bad thing, but may come as a surprise)?

File: langchain/llms/anthropic.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 class _AnthropicCommon(BaseModel):
     client: Any = None  #: :meta private:
     async_client: Any = None  #: :meta private:
-    model: str = "claude-v1"
+    model: str = "claude-v2"
```
Comment (jacoblee93) [inline langchain/llms/anthropic.py:19]: I think this needs to be `claude-2`, not v2?

**Comment by hinthornw** [comment]
> May force users to upgrade Anthropic (not necessarily a bad thing, but may come as a surprise)?

We already raise an error if it's not the right version right?

File: langchain/llms/anthropic.py (hunk 1 above)
Comment (mikelambert) [reply langchain/llms/anthropic.py:19]: Oh doh yes, thank you!

**Comment by mikelambert** [comment]
> May force users to upgrade Anthropic (not necessarily a bad thing, but may come as a surprise)?

Are you referring to "upgrade the anthropic SDK" or "update the anthropic model"?

If you're referring to model, yes, but given claude-2 has the same cost and latency as claude-1 (unlike the GPT3.5/4 bump), and given many things change between langchain versions that necessitate testing-before-deployment, it feels safe to bring everyone along here (and anyone who really cares about model stability should be explicitly pinning their model version).

**Comment by rlancemartin** [comment]
> > May force users to upgrade Anthropic (not necessarily a bad thing, but may come as a surprise)?
> 
> We already raise an error if it's not the right version right?

Hmm, @mikelambert we should confirm this. IIRC, users get a non-obvious error if they have not upgraded the Anthropic package (I recently used `Claude-2` [here](https://github.com/mendableai/QA_clustering/blob/main/notebooks/map_reduce_approach.ipynb)). But we should confirm.

**Comment by rlancemartin** [comment]
> upgrade the anthropic SDK

Yes, `upgrade the anthropic SDK`. I'm fully onboard w/ defaulting to the new model :) 

The central question is whether this breaks for users who have not upgraded anthropic SDK.

**Comment by mikelambert** [comment]
Ah gotcha! I had assumed https://github.com/hwchase17/langchain/pull/7237 would have forced users to bump to anthropic SDK 0.3.

**Comment by rlancemartin** [comment]
> Ah gotcha! I had assumed #7237 would have forced users to bump to anthropic SDK 0.3.

Oh nice, `pyproject.toml` has been updated.

**Review by rlancemartin — APPROVED** [review]

**Comment by rlancemartin** [comment]
Merging this now.

**Pull request merged**